internal/connector/connector.go

```diff
@@ -84,9 +90,9 @@ func Run(ctx context.Context, options Options) error {
 		"plaintext": secrets.NewPlainSecretProviderFromConfig(secrets.GenericConfig{}),
 	}
 
-	u, err := urlx.Parse(options.Server)
+	u, err := urlx.Parse(options.Server.URL)
 	if err != nil {
-		logging.Errorf("server: %s", err)
+		return fmt.Errorf("invalid server url: %s", err)
```

Comment:
Seems like this should be an error (not simply logging), because it will panic on the next line if we don't return here.